lingcorpora/corpora/rus_poetic_corpus.py

```diff
@@ -0,0 +1,727 @@
+                if type(value[0]) == int:
+                    subcorp += key + '=' + str(value) + '&'
+                elif key == 's_sp_frm_sch':
+                    subcorp += key + '=' + '+'.join(value[0].split()) + '&'
+                else:
+                    subcorp += key + '=' + '|'.join(value) + '&'
+        return subcorp
+
+    def __get_subcorp(self):
+        req = requests.get(self.__subcorp_url + self.__parse_sub_dict() + f'&p={self.__page}')
+        if req.status_code == 429:
+            while req.status_code == 429:
+                time.sleep(10)
+                req = requests.get(self.__subcorp_url + self.__parse_sub_dict() + f'&p={self.__page}')
+        soup = BeautifulSoup(req.content, 'html.parser')
+        return soup
+
+    def __clean_soup(self, soup):
+        """
+        A function removing NoneType lines from soup
+        """
+        if not self.__markup:
+            for elem in soup.findAll('td', class_='frm'):
+                elem.extract()
+        else:
+            for elem in soup.findAll('td', class_='frm'):
+                elem.string.replace_with('\n' + elem.text.strip() + '\t')
+
+        for elem in soup.findAll('td', width='100%'):
+            elem.replaceWithChildren()
+        for elem in soup.findAll('tr'):
+            elem.replaceWithChildren()
+        for elem in soup.findAll('br'):
+            elem.replaceWithChildren()
+        return soup
+
+    def __get_page(self, page_number, value):
+        """
+        return: etree of the page
+        """
+        headers = {
+            'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.95 Safari/537.36'}
+        if self.subcorpus is None:
+            subcorp = ''
+        else:
+            subcorp_soup = self.__get_subcorp()
+            subcorp = parse.parse_qs(parse.urlsplit(subcorp_soup.find("a", string="English")['href']).query)['mycorp'][
+                0]
+        second_part_url = ''
+        if not self.__exact:
+            if len(self.query.split()) > 1:
+                for query_item in range(2, len(self.query.split()) + 1):
+                    second_part_url += f'parent{query_item}=0&level{query_item}=0&min{query_item}=1&max{query_item}=1&lex{query_item}={self.query.split()[query_item - 1]}&gramm=&sem-mod{query_item}=sem&sem-mod{query_item}=semx&sem{query_item}=&form{query_item}=&flags{query_item}=&'
+        req = requests.get(self.__url.format(subcorp, value, self.query.split()[0], page_number) + second_part_url,
+                           headers=headers)
+        soup = BeautifulSoup(req.content, 'html.parser')
+        self.__clean_soup(soup)
+        return soup
+
+    def __ana(self, source):
+        """
+        A word's analysis parser
+        """
+        ana = {}
+
+        time.sleep(randint(0, 3))
+        req = requests.get(self.__ana_url + source)
+        if req.status_code == 429:
+            while req.status_code == 429:
```

Comment:
it can return `429` all the time, which would result in an infinite loop. Could you please add a setting with a maximum number of retries and/or a maximum period of waiting